crates/algokit_utils/src/transactions/composer.rs

```diff
@@ -449,32 +452,257 @@ impl Composer {
+            return Err(ComposerError::GroupSizeError());
+        }
+
+        for composer_transaction in composer_transactions {
+            self.push(composer_transaction)?;
+        }
+
+        Ok(())
+    }
+
+    pub fn add_app_call_method_call(
         &mut self,
-        app_call_params: ApplicationCallParams,
+        params: AppCallMethodCallParams,
     ) -> Result<(), ComposerError> {
-        self.push(ComposerTransaction::ApplicationCall(app_call_params))
+        self.add_app_method_call_internal(&params.args, || {
+            ComposerTransaction::AppCallMethodCall((&params).into())
+        })
     }
 
-    pub fn add_application_create(
+    pub fn add_app_create_method_call(
         &mut self,
-        app_create_params: ApplicationCreateParams,
+        params: AppCreateMethodCallParams,
     ) -> Result<(), ComposerError> {
-        self.push(ComposerTransaction::ApplicationCreate(app_create_params))
+        self.add_app_method_call_internal(&params.args, || {
+            ComposerTransaction::AppCreateMethodCall((&params).into())
+        })
     }
 
-    pub fn add_application_update(
+    pub fn add_app_update_method_call(
         &mut self,
-        app_update_params: ApplicationUpdateParams,
+        params: AppUpdateMethodCallParams,
     ) -> Result<(), ComposerError> {
-        self.push(ComposerTransaction::ApplicationUpdate(app_update_params))
+        self.add_app_method_call_internal(&params.args, || {
+            ComposerTransaction::AppUpdateMethodCall((&params).into())
+        })
     }
 
-    pub fn add_application_delete(
+    pub fn add_app_delete_method_call(
         &mut self,
-        app_delete_params: ApplicationDeleteParams,
+        params: AppDeleteMethodCallParams,
     ) -> Result<(), ComposerError> {
-        self.push(ComposerTransaction::ApplicationDelete(app_delete_params))
+        self.add_app_method_call_internal(&params.args, || {
+            ComposerTransaction::AppDeleteMethodCall((&params).into())
+        })
+    }
+
+    fn parse_abi_return_values(
+        &self,
+        confirmations: &[PendingTransactionResponse],
+    ) -> Vec<Option<Result<ABIReturn, ComposerError>>> {
+        let mut abi_returns = Vec::new();
+
+        for (i, confirmation) in confirmations.iter().enumerate() {
+            if let Some(transaction) = self.transactions.get(i) {
+                if let Some(method) = self.get_method_from_transaction(transaction) {
+                    let abi_return = self.extract_abi_return_from_logs(confirmation, method);
+                    abi_returns.push(abi_return);
+                }
+            }
+        }
+
+        abi_returns
+    }
+
+    fn extract_abi_return_from_logs(
+        &self,
+        confirmation: &PendingTransactionResponse,
+        method: &ABIMethod,
+    ) -> Option<Result<ABIReturn, ComposerError>> {
+        // Check if method has return type
+        method.returns.as_ref()?;
```

Comment:
```suggestion
        let return_type = method.returns.as_ref()?;
```
Then you can remove the `let return_type = method.returns.as_ref().unwrap(); // Safe unwrap since we checked above` later on.